ner_v2/detectors/numeral/number/standard_number_detector.py

```diff
@@ -0,0 +1,247 @@
+                                     self._detect_number_from_numerals
+                                     ]
+
+    def detect_number(self, text):
+        self.text = text
+        self.processed_text = text
+        self.tagged_text = text
+
+        number_list, original_list = None, None
+        for detector in self.detector_preferences:
+            number_list, original_list = detector(number_list, original_list)
+            self._update_processed_text(original_list)
+        return number_list, original_list
+
+    @staticmethod
+    def _get_list_from_pipe_sep_string(numerals_text):
+        """
+        Split numerals
+        Args:
+            numerals_text (str): numerals text
+        Returns:
+            (list) : list containing numeral after split
+        """
+        numerals_list = numerals_text.split("|")
+        return [num.lower().strip() for num in numerals_list if num.strip()]
+
+    def init_regex_and_parser(self, data_directory_path):
+        """
+        Initialise numbers word from from data file
+        Args:
+            data_directory_path (str): path of data folder for given language
+        Returns:
+            None
+        """
+        # create number_words dict having number variants and their corresponding scale and increment value
+        # create language_scale_map dict having scale variants and their corresponding value
+        numeral_df = pd.read_csv(os.path.join(data_directory_path, NUMBER_NUMERAL_CONSTANT_FILE), encoding='utf-8')
+        for index, row in numeral_df.iterrows():
+            name_variants = self._get_list_from_pipe_sep_string(row[NUMBER_DATA_FILE_NAME_VARIANTS])
+            value = row[NUMBER_DATA_FILE_VALUE]
+            if float(value).is_integer():
+                value = int(row[NUMBER_DATA_FILE_VALUE])
+            number_type = row[NUMBER_DATA_FILE_TYPE]
+
+            if number_type == NUMBER_TYPE_UNIT:
+                for numeral in name_variants:
+                    self.numbers_word[numeral] = (1, value)
+            else:
+                for numeral in name_variants:
+                    self.numbers_word[numeral] = (value, 0)
+                    self.language_scale_map[numeral] = value
+
+        # create units_dict having unit variants and their corresponding value
+        unit_file_path = os.path.join(data_directory_path, NUMBER_UNITS_FILE)
+        if os.path.exists(unit_file_path):
+            units_df = pd.read_csv(unit_file_path, encoding='utf-8')
+            for index, row in units_df.iterrows():
+                unit_variants = self._get_list_from_pipe_sep_string(row[NUMBER_UNIT_VARIANTS])
+                unit_value = row[NUMBER_UNIT_VALUE]
+                for unit in unit_variants:
+                    self.units_map[unit] = unit_value
+
+    def _get_unit_from_text(self, detected_original, processed_text):
+        """
+        Method to detect unit from detected original text.
```

Comment:
Better description of logic needed for core functions '_get_unit_from_text', '_detect_number_from_numerals' nd '_detect_number_from_digit'